Sources/ConfidenceProvider/ConfidenceFeatureProvider.swift

```diff
@@ -260,43 +260,46 @@ public class ConfidenceFeatureProvider: FeatureProvider {
             throw OpenFeatureError.invalidContextError
         }
 
-        do {
-            let resolverResult = try resolver.resolve(flag: path.flag, ctx: ctx)
-
-            guard let value = resolverResult.resolvedValue.value else {
-                return resolveFlagNoValue(
-                    defaultValue: defaultValue,
-                    resolverResult: resolverResult,
-                    ctx: ctx
-                )
-            }
+        let resolverResult = try resolver.resolve(flag: path.flag, ctx: ctx)
 
-            let pathValue: Value = try getValue(path: path.path, value: value)
-            guard let typedValue: T = pathValue == .null ? defaultValue : pathValue.getTyped() else {
-                throw OpenFeatureError.parseError(message: "Unable to parse flag value: \(pathValue)")
-            }
-
-            let evaluationResult = ProviderEvaluation(
-                value: typedValue,
-                variant: resolverResult.resolvedValue.variant,
-                reason: Reason.targetingMatch.rawValue
+        guard let value = resolverResult.resolvedValue.value else {
+            return resolveFlagNoValue(
+                defaultValue: defaultValue,
+                resolverResult: resolverResult,
+                ctx: ctx
             )
+        }
 
+        let pathValue: Value = try getValue(path: path.path, value: value)
+        guard let typedValue: T = pathValue == .null ? defaultValue : pathValue.getTyped() else {
+            throw OpenFeatureError.parseError(message: "Unable to parse flag value: \(pathValue)")
+        }
+
+        if resolverResult.resolvedValue.resolveReason == .stale {
             processResultForApply(
```

Comment:
why do we handle this differently from other cases and why [here](https://github.com/spotify/confidence-openfeature-provider-swift/pull/98/files#diff-1bad310998a7169fc07cc21c6d84c6ff339070d2559f7c3406b2dfc9fc76dc08R346) we are returning the default value? can we handle this stale case also like the others?

Reply:
i got the answer for the second question i asked, the second function is for default values. however we can still go on without this condition block and only update the reason if needed in [this line ](https://github.com/spotify/confidence-openfeature-provider-swift/pull/98/files#diff-1bad310998a7169fc07cc21c6d84c6ff339070d2559f7c3406b2dfc9fc76dc08R300)